libs/muddle/src/direct_message_service.cpp

```diff
@@ -420,16 +425,10 @@ DirectMessageService::UpdateStatus DirectMessageService::UpdateReservation(Addre
       auto conn = register_.LookupConnection(handle).lock();
       if (conn)
       {
-        // determine the connection orientation
-        bool const is_outgoing = conn->Type() == network::AbstractConnection::TYPE_OUTGOING;
-
         // we have a duplicate entry. This means that the two nodes are trying to connect to each
         // other. In this case define the rule that the connection with the lowest address continues
         // the connection
-        bool const should_replace =
-            ((address_ < address) && is_outgoing) || ((address_ > address) && !is_outgoing);
-
-        if (should_replace)
+        if (ShouldReplace(address, IsOutgoing(conn)))
```

Comment:
Yes that is true, but would require passing the connection object into `ShouldReplace`. I don't really see much advantage in either case to be honest.

Reply:
Well, if `is_outgoing` was computed in `ShouldReplace()`, the call itself would be shorter, and with less parentheses; and if `IsOutgoing()` is not used anywhere, it could be eliminated at all.

Reply:
I am a little surprised at the level of discussion around this point. The changes were originally made  to aid in some additional changes in this file which proved unnecessary.

For the time being I have reverted all these changes back to the original